clients/include/device_batch_vector.hpp

```diff
@@ -208,9 +208,7 @@ class device_batch_vector : private d_vector<T, PAD, U>
     //!
     hipError_t memcheck() const
     {
-        if(!m_enough_memory)
-            return hipErrorMemoryAllocation;
-        else if((bool)*this)
+        if((bool)*this)
```

Comment:
`(bool)` cast is not necessary, since when used in `if(...)` or other boolean contexts, the argument is converted to `bool`, even if only an `explicit` conversion to `bool` exists. `if(*this)` is sufficient.